WordPress/gutenberg
# Image: Display custom borders on placeholder

Fixes: https://github.com/WordPress/gutenberg/issues/49144

Related:
- https://github.com/WordPress/gutenberg/pull/42847

## What?

Applies custom borders from either Global Styles or individual border block supports to the Image block's placeholder when it isn't selected.

## Why?

This brings the Image block more in line with the Post Featured Image block which displays borders on its placeholder.

## How?

- Updates the Image block's selectors to include the placeholder component class for the border support
- Adds the border classes and inline styles to media placeholder component when the block isn't selected


## Testing Instructions

1. Edit a post, add an image block but do not choose an image.
2. Deselect the block and check the media placeholder displays correctly.
3. Switch to the site editor and add a border to the Image block via Global Styles (or add one via theme.json).
4. Switch back to your post and reload
5. Without selecting the image block you should see the border you set
6. Select the block and confirm that no border is displayed
7. In the block inspector, configure a custom border just for this Image block
8. Deselect the block and make sure your border is displayed correctly
9. Select the block and choose an image
10. Confirm your border continues to display correctly on the block now it has an image.

## Screenshots or screencast 

https://user-images.githubusercontent.com/60436221/229733475-dbd77307-277c-4191-aa9c-d65ae2086b48.mp4
